01_prepare_host.sh

```diff
@@ -223,24 +223,24 @@ popd
 
 # NOTE(elfosardo): workaround for https://github.com/moby/moby/issues/44970
 # should be fixed in docker-ce 23.0.2
-if [[ $OS == ubuntu ]]; then
+if [ ${OS} = "ubuntu" ]; then
   sudo systemctl restart docker
 fi
 
 # Pulling all the images except any local image.
 for IMAGE_VAR in $(env | grep -v "_LOCAL_IMAGE=" | grep "_IMAGE=" | grep -o "^[^=]*") ; do
   IMAGE="${!IMAGE_VAR}"
-  pull_container_image_if_missing "$IMAGE"
+  pull_container_image_if_missing "${IMAGE}"
  done
 
-if ${IPA_DOWNLOAD_ENABLED} || [ ! -f "${IRONIC_DATA_DIR}/html/images/ironic-python-agent.kernel" ]; then
+if [ "${IPA_DOWNLOAD_ENABLED}" = "true" ] || [ ! -e "${IRONIC_DATA_DIR}/html/images/ironic-python-agent.kernel" ]; then
```

Comment:
Technically `-f` (file exists) is more correct than `-e` (something exists), but the `-r` (exists and readable by user) is the best.